src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -44,9 +48,15 @@ void uct_rdmacm_cm_ep_error_cb(uct_rdmacm_cm_ep_t *cep,
                                uct_cm_remote_data_t *remote_data,
                                ucs_status_t status)
 {
-    if (cep->flags & UCT_RDMACM_CM_EP_CONNECTED) {
+    if (cep->flags & UCT_RDMACM_CM_EP_FAILED) {
+        return;
+    }
+
+    cep->flags |= UCT_RDMACM_CM_EP_FAILED;
+    cep->status = status;
```

Comment:
can we move assert from line 61 to here. and rewrite lines 60-68 as ".. else if ( ... "

Reply:
done